### PR TITLE
Cover resumption with unit tests

### PR DESCRIPTION
**Pull request by Kozoriz**
- SQLDatabase class divided  to interface and impl.
- Created mock for SQLDatabase 
- Added tests for resumptionDataDB, resumptionDataJson, resumptionCtrl classes

Pull request was reviewed in : https://github.com/smartdevicelink/sdl_core/pull/618

### CODE REVIEW

**Comment by Kozoriz** [comment]
This pull was reviewed in #618 
Please review : @anosach-luxoft , @dev-gh , @VProdanov , @LevchenkoS, @okozlovlux

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 1]
```diff
@@ -138,15 +141,15 @@ TEST_F(ResumeCtrlTest, StartResumption_AppWithGrammarId) {
...
+  EXPECT_CALL(mock_app_mngr_, GetDefaultHmiLevel(_))
       .WillRepeatedly(Return(kDefaultTestLevel_));
   EXPECT_CALL(*mock_storage,
               GetSavedApplication(kTestPolicyAppId_, kMacAddress_, _))
       .WillRepeatedly(DoAll(SetArgReferee<2>(saved_app), Return(true)));
   EXPECT_CALL(*app_mock, UpdateHash());
   EXPECT_CALL(*app_mock, set_grammar_id(kTestGrammarId_));
 
-  bool res = res_ctrl->StartResumption(app_mock, kHash_);
+  const bool res = res_ctrl->StartResumption(app_mock, kHash_);
```
Comment (LevchenkoS) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz, wrong const naming, please check everywhere.

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 2]
```diff
@@ -469,25 +472,44 @@ TEST_F(ResumeCtrlTest, StartResumption_AppWithSubscriptionToIVI) {
...
 
   EXPECT_CALL(*app_mock, UpdateHash());
-  bool res = res_ctrl->StartResumption(app_mock, kHash_);
+  const bool res = res_ctrl->StartResumption(app_mock, kHash_);
+  EXPECT_TRUE(res);
+}
+
+TEST_F(ResumeCtrlTest, StartResumption_AppWithSubscriptionToWayPoints) {
+  smart_objects::SmartObject saved_app;
+  saved_app[application_manager::strings::hash_id] = kHash_;
```
Comment (LevchenkoS) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz, Don't it be better like this `am::strings::hash_id`?

File: src/components/utils/test/sqlite_wrapper/sql_query_test.cc [hunk 3]
```diff
@@ -102,7 +102,7 @@ const std::string SQLQueryTest::kDatabaseName = "test-query";
 TEST_F(SQLQueryTest, Query_CreateQuery_QueryInDBEqualCreated) {
   // arrange
   const std::string kSelect("SELECT * FROM testTable WHERE integerValue = ?");
-  SQLDatabase db(kDatabaseName, "test");
+  SQLDatabaseImpl db(kDatabaseName);
```
Comment (LevchenkoS) [inline src/components/utils/test/sqlite_wrapper/sql_query_test.cc]: @Kozoriz, Can you move `db` variable to the fixture class?

**Comment by LevchenkoS** [comment]
@Kozoriz, reviewed.

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 2 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @LevchenkoS , no, if line shorter then 80, let it be. + This more readable.

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 4]
```diff
@@ -686,27 +765,44 @@ TEST_F(ResumeCtrlTest, IsHMIApplicationIdExist) {
...
+  const std::string kPolicyAppId = "policy_app_id";
+  const std::string kDeviceId = "device_id";
+
+  EXPECT_CALL(*mock_storage, IsApplicationSaved(kPolicyAppId, kDeviceId))
+      .WillOnce(Return(0));
+  res_ctrl->OnAppRegistrationStart(kPolicyAppId, kDeviceId);
+}
+
+TEST_F(ResumeCtrlTest, OnAppRegistrationEnd) {
+  uint32_t timeout = 10u;
```
Comment (okozlovlux) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz can timeout be const?

File: src/components/application_manager/test/CMakeLists.txt [hunk 5]
```diff
@@ -178,10 +178,10 @@
 
     # TODO [AKozoriz] : Fix not buildable tests
     set(ResumptionData_SOURCES
```
Comment (dev-gh) [inline src/components/application_manager/test/CMakeLists.txt]: @Kozoriz 
Looks like you can remove that comment now

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 6]
```diff
@@ -567,80 +589,137 @@ TEST_F(ResumeCtrlTest, SetupDefaultHMILevel) {
...
+      .Times(AtLeast(1));
+  GetInfoFromApp();
+  EXPECT_CALL(*mock_storage, GetSavedApplication(kTestPolicyAppId_, _, _))
+      .Times(2)
+      .WillRepeatedly(DoAll(SetArgReferee<2>(saved_app), Return(true)));
+
+  EXPECT_CALL(*mock_storage, RemoveApplicationFromSaved(_, _))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(mock_app_mngr_, GetUserConsentForDevice("12345"))
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz 
Looks like this device id is set in some other place, so maybe it worth to use variable instead?

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 7]
```diff
@@ -686,27 +765,44 @@ TEST_F(ResumeCtrlTest, IsHMIApplicationIdExist) {
...
+  uint32_t timeout = 10u;
+  EXPECT_CALL(mock_application_manager_settings_,
+              app_resumption_save_persistent_data_timeout())
+      .WillOnce(ReturnRef(timeout));
+  res_ctrl->OnAppRegistrationEnd();
 }
 
 TEST_F(ResumeCtrlTest, CheckPersistenceFiles_WithoutCommandAndChoiceSets) {
-  uint32_t ign_off_count = 0;
+  const uint32_t kIggnOffCount = 0;
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz 
Variable name has typo

File: src/components/application_manager/test/resumption/resumption_data_json_test.cc [hunk 8]
```diff
@@ -78,37 +79,35 @@ class ResumptionDataJsonTest : public ResumptionDataTest {
...
     }
   }
 
   void SetZeroIgnOff() {
-    Value& dictionary = last_state_.dictionary;
-    Value& res = dictionary[am::strings::resumption];
-    res[am::strings::last_ign_off_time] = 0;
+    utils::json::JsonValue& dictionary = last_state_.dictionary();
+    utils::json::JsonValueRef res = dictionary[am::strings::resumption];
+    res[am::strings::last_ign_off_time] = 0ll;
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resumption_data_json_test.cc]: @Kozoriz 
Method name inconsistent with logic

**Comment by dev-gh** [comment]
Reviewed

File: src/components/application_manager/test/resumption/resumption_data_json_test.cc (hunk 8 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_json_test.cc]: @dev-gh , done 163b6d5

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 7 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @dev-gh , done 163b6d5

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 6 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @dev-gh , done 163b6d5

File: src/components/application_manager/test/CMakeLists.txt (hunk 5 above)
Comment (Kozoriz) [inline src/components/application_manager/test/CMakeLists.txt]: @dev-gh , done 163b6d5

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 4 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @okozlovlux done 163b6d5

File: src/components/utils/test/sqlite_wrapper/sql_query_test.cc (hunk 3 above)
Comment (Kozoriz) [inline src/components/utils/test/sqlite_wrapper/sql_query_test.cc]: @LevchenkoS , done 163b6d5

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 1 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @LevchenkoS . done 163b6d5

**Comment by okozlovlux** [comment]
@Kozoriz reviewed

**Comment by dev-gh** [comment]
Reviewed

**Comment by VProdanov** [comment]
@Kozoriz Reviewed, just fix the build fails.

**Comment by anosach-luxoft** [comment]
Reviewed

**Pull request merged**